aiohttp/base_protocol.py

```diff
@@ -81,7 +81,7 @@ async def _drain_helper(self) -> None:
         if not self._paused:
             return
         waiter = self._drain_waiter
-        assert waiter is None or waiter.cancelled()
-        waiter = self._loop.create_future()
-        self._drain_waiter = waiter
-        await waiter
+        if waiter is None:
+            waiter = self._loop.create_future()
+            self._drain_waiter = waiter
+        await asyncio.shield(waiter)
```

Comment:
Why is `shield` required?
If the connection is closed by another task -- the current task should get the exception as well.